VideoPress: Use camera icon as thumbnail on private videos

lhkowalski opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Relates to #26675.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* On the VideoPress videos, show a camera icon as the thumbnail when the video is private; this will prevent the `403` response when trying to load the original thumbnail, that is private in this case
* The change works on both the grid and list view
* Note: this will not solve the problem inside the video details page (I am planning another PR and another approach inside that page)

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to `Jetpack > VideoPress` to see the grid of videos 
* Set the privacy for some video to "Private":

![image](https://user-images.githubusercontent.com/6760046/196524103-556f0aae-ee62-47b2-811d-cefdedcfb9b1.png)

* Confirm that now it shows a camera icon as the thumbnail:

![image](https://user-images.githubusercontent.com/6760046/196524219-9870a87b-2651-4b48-9921-f9fa6a5e8c00.png)

* Confirm that every other private video is showing the same camera icon as well:

![image](https://user-images.githubusercontent.com/6760046/196524320-e6057c92-b2b8-4862-949d-625a57d2a0a0.png)

* Confirm that the same happens when you switch the view from grid to list:

![image](https://user-images.githubusercontent.com/6760046/196524442-41836fca-0e08-48be-904f-7e535f34e05f.png)

lhkowalski committed:
Use camera icon thumbnail when video privacy is set to private

lhkowalski committed:
Set privacy setting prop on video card inside video grid

lhkowalski committed:
Add changelog file

retrofox commented on projects/packages/videopress/src/client/admin/components/video-card/index.tsx:
```diff
@@ -62,6 +63,12 @@ const ProcessingThumbnail = () => (
 	</div>
 );
 
+const PrivateThumbnail = () => (
+	<div className={ styles[ 'video-card__custom-thumbnail' ] }>
+		<Icon icon={ captureVideo } size={ 96 } />
+	</div>
+);
```
I wonder whether creating a new component does make sense, especially considering that you are not taking advantage of the CSS class. I guess it was part of your development flow which is fine. Just consider removing it and using straight `<Icon icon={ captureVideo } size={ 96 } />`. BTW it looks pretty nice.

<img width="990" alt="image" src="https://user-images.githubusercontent.com/77539/196303022-d5914e3a-eea9-4bd9-a485-146f4c89dd9f.png">

lhkowalski replied:
@retrofox This is what I get when using only the icon:

![image](https://user-images.githubusercontent.com/6760046/196494515-b9e87b28-f7b0-4312-8688-2c17455e0686.png)

Seems like I do need this `<div className={ styles[ 'video-card__custom-thumbnail' ] }>` wrapper to make the style works properly.

Is it OK to use it inline to prevent the component creation? Like this:

```
	thumbnail = privacyIsPrivate ? (
		<div className={ styles[ 'video-card__custom-thumbnail' ] }>
			<Icon icon={ captureVideo } size={ 96 } />
		</div>
	) : thumbnail;
```

retrofox commented on projects/packages/videopress/src/client/admin/components/video-grid/index.tsx:
```diff
@@ -32,6 +32,7 @@ const VideoGrid = ( { videos, count = 6, onVideoDetailsClick, loading }: VideoGr
 							<VideoCard
 								id={ video?.id }
 								title={ video.title }
+								privacySetting={ video.privacySetting }
```
Although we pass some properties to the `<Video Card />` component, we should not. The only required in terms of getting data is the video ID. The other ones are provided by the `useVideo()` hook, called adequately by the component.

[This is](https://github.com/Automattic/jetpack/blob/f45e61deec79634bafeb8c78b6413be6fb1b5ec4/projects/packages/videopress/src/client/admin/components/video-card/index.tsx#L181) the lin where it calls the hook. It provides some state props, but even is possible to get all video data:

```es6
const { data: video } = useVideo( <video-id> );
```

You can take a look at the [VideoQuickActions component](https://github.com/Automattic/jetpack/blob/f45e61deec79634bafeb8c78b6413be6fb1b5ec4/projects/packages/videopress/src/client/admin/components/video-quick-actions/index.tsx#L241) to get the idea.

I suggest not passing the `privacySetting` prop down and instead picking it up straight from [the ConnectVideoCard component](https://github.com/Automattic/jetpack/blob/f45e61deec79634bafeb8c78b6413be6fb1b5ec4/projects/packages/videopress/src/client/admin/components/video-card/index.tsx#L180-L197).

lhkowalski replied:
@retrofox Related to this, we are not using the `ConnectVideoCard` component (it was declared but never used). Is it OK to use the cutom hook inside the `VideoCard` itself, or it violates our conventions?

Alternativelly, I can use this opportunity to really wire the `ConnectVideoCard` replacinf the `VideoCard` itself. What to you think?

retrofox replied:
> ...we are not using the `ConnectVideoCard` component 

I think we are. ConnectVideoCard is the [default function exported by the file](https://github.com/Automattic/jetpack/blob/3f822142dd9c1a692674a41925dd3c62a921d989/projects/packages/videopress/src/client/admin/components/video-card/index.tsx#L199)

:-)

lhkowalski replied:
Oh, now I see! :man_facepalming:

lhkowalski replied:
Tried the change here, was able to use it from the `data` available on the `useVideo` hook. Nice!

lhkowalski committed:
Remove custom component to use inline markup

lhkowalski committed:
Remove privacySetting prop in favor of useVideo hook usage

lhkowalski committed:
Refactor thumbnail rendering to organize thumbnail choices

lhkowalski committed:
Add support for private thumbnails on the VideoThumbnail compoment

When the thumbnail is set to private, we will show a placeholder with a camera icon inside it.

lhkowalski committed:
Use the video privacySetting to decide if the video thumbnail may be private

lhkowalski committed:
Fix typo on changelog file

lhkowalski committed:
Change VideoCard component to use thumbnail isPrivate prop

Additionally, use the video privacy setting to decide which value to set on the isPrivate prop.

lhkowalski committed:
Merge branch 'trunk' into fix/use-camera-icon-as-thumbnail-on-private-videos

dhasilva reviewed (approved):
The edit page, specially the `Select from video` still has issues. And when setting the privacy back to `public` the image did not load here. We can tackle that on a follow-up. Nice work!
